pkg/kubernetes/kubernetes.go

```diff
@@ -26,23 +29,51 @@ const (
 
 type CloseWatchKubeConfig func() error
 
-type Kubernetes struct {
+type Kubernetes interface {
+	WatchKubeConfig(onKubeConfigChange func() error)
+	Close()
+	Derived(ctx context.Context) DerivedKubernetes
+	ConfigurationView(minify bool) (runtime.Object, error)
+	IsOpenShift(ctx context.Context) bool
+}
+
+type DerivedKubernetes interface {
```

Comment:
This interface is too rigid to be used to implement different `DerivedKubernetes` objects. In my opinion, having a struct is better, probably after renaming the `Kubernetes` (which hides the package names) to `KubernetesManager`.